server/controllers/logsController.ts

```diff
@@ -0,0 +1,155 @@
+import type { Request, Response, NextFunction } from 'express';
+import type createLogsService from './services/logsService';
+import { InternalServerError } from '@/utils/customErrors';
+import { readdir, readFileSync, unlinkSync } from 'fs';
+import { Readable } from 'stream';
+import { promisify } from 'util';
+import { Binary } from 'mongodb';
+import archiver from 'archiver';
+import log4js from 'log4js';
+
+export interface LogFile {
+  filename: string;
+  fileData: Buffer;
+}
+
+const logger = log4js.getLogger('[LOGS CONTROLLER]');
+const readdirAsync = promisify(readdir);
+
+const createLogsController = (
+  logsService: ReturnType<typeof createLogsService>,
+) => {
+  const privateLogsService = logsService;
+  const retentionCutoffDate = () => {
+    // 10 days ago
+    return new Date(Date.now() - 10 * 24 * 60 * 60 * 1000);
+    // 2 seconds ago for testing purposes:
+    // return new Date(Date.now() - 2 * 1000);
+  };
+
+  // upload the cold log files from the local /logs directory to the database
+  const uploadLogs = async () => {
+    try {
+      // Read the logs directory
+      const files = await readdirAsync('./logs');
+
+      logger.debug('Uploading log files:', files);
+
+      // Process each log file
+      const logsUpdate: LogFile[] = files.map((filename) => {
+        const filePath = `./logs/${filename}`;
+        const fileData = readFileSync(filePath);
+        return { filename, fileData };
+      });
+
+      // Upload the logs to the database
+      await logsService.uploadLogs(logsUpdate);
+    } catch (error) {
+      logger.error('Error processing logs:', error);
+    }
+  };
+
+  const evictLogs = async () => {
+    // delete aged logs from the local directory
+    try {
+      // Read the logs directory
+      const files = await readdirAsync('./logs');
+
+      logger.debug('Evicting log files:', files);
+
+      for (const filename of files) {
+        // skip the hot log file
+        if (filename === 'linkta-server.log') return;
```

Comment:
Do you mean to exit the for loop or it was a typo (maybe you meant to use `continue`)?

Reply:
I think there may have been an earlier revision that was built around a forEach() method, and the return was there to skip the cycle.